examples/ecs/run_conditions.rs

```diff
@@ -0,0 +1,42 @@
+//! This example demonstrates how to use run criterias to control when systems run.
+
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<InputCounter>()
+        .add_system(
+            increment_input_counter
+                .run_if(resource_exists::<InputCounter>())
+                .run_if(is_input),
+        )
+        .add_system(
+            print_input_counter
+                .run_if(resource_exists::<InputCounter>())
+                .run_if(|c: Res<InputCounter>| c.is_changed() && !c.is_added()),
```

Comment:
This should have a comment explaining that this is a closure, and why it meets the `run_if` signature.